api/src/main/java/org/apache/iceberg/DeleteFiles.java

```diff
@@ -81,4 +81,12 @@ default DeleteFiles deleteFile(DataFile file) {
    * @return this for method chaining
    */
   DeleteFiles caseSensitive(boolean caseSensitive);
+
+  /**
+   * Enables validation that any files that are part of the deletion still exist when committing the
+   * operation.
+   *
+   * @return this for method chaining
+   */
+  DeleteFiles validateFilesExist();
```

Comment:
Instead of making an exception in revapi, we need to add a default implementation of this method that throws `UnsupportedOperationException`. That avoids the breaking API change and is fine because we don't expect anyone to be implementing these methods. In any case, if an implementation does exist then throwing is the right thing to do.

Reply:
Good point, we can totally avoid breaking the `DeleteFile`  API change. anyone who happens to be implementing this isn't forced to override it until they actually need a validation behavior. Fixed